src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -57,28 +61,38 @@ export function useTextureData(): Uint8Array | undefined {
   );
 
   const values = useValues();
-
   const { proxy } = useComlink<TextureWorker>(() => new Worker(), []);
-  const [textureData, setTextureData] = useState<Uint8Array>();
+
+  const [state, mergeState] = useSetState<TextureDataState>({});
```

Comment:
`useSetState` provides a function to merge the state instead of just overriding it completely. It simplifies setting `loading` to `true` while keeping the existing texture data (to avoid a flash).